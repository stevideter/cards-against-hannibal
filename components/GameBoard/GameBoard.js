import styles from "./GameBoard.module.css";
import BlackCard from "../BlackCard/BlackCard";
import WhiteCard from "../WhiteCard/WhiteCard";

const GameBoard = () => (
  <div className={styles.gameboard}>
    <div className={styles.blackcard}>
    <BlackCard />
    </div>
    <div className={styles.whitecards}>
      <WhiteCard text="design" />
      <WhiteCard text="murder" />
      <WhiteCard text="fish jello" />

    </div>
  </div>
);

export default GameBoard;
