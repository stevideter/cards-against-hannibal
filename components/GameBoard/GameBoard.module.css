.gameboard {
    background-color: white;   
}
.blackcard {
    margin: 10px;

}
.whitecards {
    margin: 10px;
    display: flex;
    flex-direction:row;
}