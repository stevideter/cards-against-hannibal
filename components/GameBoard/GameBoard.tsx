import styled from 'styled-components';
import BlackCard from '../BlackCard/BlackCard';
import WhiteCard from '../WhiteCard/WhiteCard';

const StyledGameBoard = styled.div`
    background-color: white;
`;
const StyledBlackCard = styled.div`
    margin: 10px;
`;
const StyledWhiteCards = styled.div`
    margin: 10px;
    display: flex;
    flex-direction: row;
`;
interface GameBoardProps {
    whiteCards: Card[];
    blackCards: Card[];
}

const GameBoard = (props: GameBoardProps): JSX.Element => {
    const whiteCards = [];
    const fullDeck = [...props.whiteCards];
    let cardCount = props.whiteCards.length;
    for (let i = 0; i < 10; i++) {
        const randomIndex = Math.floor(Math.random() * cardCount);
        const card = fullDeck.splice(randomIndex, 1)[0];
        whiteCards.push(<WhiteCard text={card.text} key={i} id={card.id} />);
        cardCount--;
    }
    const randomIndex = Math.floor(Math.random() * props.blackCards.length);

    const blackCard = <BlackCard text={props.blackCards[randomIndex].text} />;
    return (
        <StyledGameBoard>
            <StyledBlackCard>{blackCard}</StyledBlackCard>
            <StyledWhiteCards>{whiteCards}</StyledWhiteCards>
        </StyledGameBoard>
    );
};

export default GameBoard;
