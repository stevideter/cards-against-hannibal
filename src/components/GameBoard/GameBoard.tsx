import styled from 'styled-components';
import BlackCard from '../BlackCard/BlackCard';
import WhiteCard from '../WhiteCard/WhiteCard';

const StyledGameBoard = styled.div`
    background-color: white;
`;
const StyledBlackCard = styled.div`
    margin: 10px;
`;
const StyledWhiteCards = styled.div`
    margin: 10px;
    display: flex;
    flex-direction: row;
`;
interface GameBoardProps {
    whiteCards: Card[];
    blackCards: Card[];
}

const GameBoard: React.FunctionComponent<GameBoardProps> = (
    props: GameBoardProps
) => {
    const { whiteCards } = props;
    const whiteCardList = whiteCards.map((card, i) => (
        <WhiteCard text={card.text} key={i} id={card.id} />
    ));
    const randomIndex = Math.floor(Math.random() * props.blackCards.length);

    const blackCard = <BlackCard text={props.blackCards[randomIndex].text} />;
    return (
        <StyledGameBoard>
            <StyledBlackCard>{blackCard}</StyledBlackCard>
            <StyledWhiteCards>{whiteCardList}</StyledWhiteCards>
        </StyledGameBoard>
    );
};

export default GameBoard;
