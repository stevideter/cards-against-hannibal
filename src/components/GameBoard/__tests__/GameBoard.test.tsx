jest.mock('../../BlackCard/BlackCard', () => 'BlackCard');
jest.mock('../../WhiteCard/WhiteCard', () => 'WhiteCard');

import GameBoard from '../GameBoard';
import renderer from 'react-test-renderer';

describe('Game', () => {
    it('renders', () => {
        const whiteCards: Card[] = Array(10)
            .fill(0)
            .reduce((arr) => {
                arr.push(arr.length);
                return arr;
            }, [])
            .map((num: unknown) => ({ id: num, text: 'text' }));
        const currentRound: Round = {
            count: 1,
            blackCard: { id: 'id', text: 'text', blanks: 1 },
            whiteCards: [],
        };
        const tree = renderer
            .create(
                <GameBoard
                    whiteCards={whiteCards}
                    currentRound={currentRound}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
