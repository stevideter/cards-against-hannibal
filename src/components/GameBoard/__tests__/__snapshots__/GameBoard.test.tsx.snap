// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders 1`] = `
<div
  className="sc-AxjAm fhsYJt"
>
  <div
    className="sc-AxiKw fkSgHu"
  >
    <div
      className="sc-AxirZ geUgcE"
    >
      <BlackCard
        text="text"
      />
    </div>
    <div
      className="sc-AxhCb reGgE"
    />
  </div>
  <div
    className="sc-AxhCb reGgE"
  >
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
    <WhiteCard
      dispatch={[Function]}
      id="id"
      text="text"
    />
  </div>
</div>
`;
